File: backend-go/internal/modules/system/entity/sys_menu.go
package entity

import (
	"drip-admin/backend-go/internal/common"
	"time"
)

type SysMenu struct {
	ID             common.Int64String `gorm:"column:id;primaryKey" json:"id"`
	ParentID       common.Int64String `gorm:"column:parent_id" json:"parentId"`
	Name           string             `gorm:"column:name" json:"name"`
	Type           string             `gorm:"column:type" json:"type"`
	Path           *string            `gorm:"column:path" json:"path"`
	Component      *string            `gorm:"column:component" json:"component"`
	PermissionCode *string            `gorm:"column:permission_code" json:"permissionCode"`
	Icon           *string            `gorm:"column:icon" json:"icon"`
	Sort           int                `gorm:"column:sort" json:"sort"`
	Visible        int                `gorm:"column:visible" json:"visible"`
	Status         int                `gorm:"column:status" json:"status"`
	Deleted        int                `gorm:"column:deleted" json:"deleted"`
	CreatedAt      *time.Time         `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt      *time.Time         `gorm:"column:updated_at" json:"updatedAt"`
}

func (SysMenu) TableName() string { return "sys_menu" }

File: backend-python/app/modules/system/entity/sys_menu.py
from sqlalchemy import BigInteger, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.modules.system.entity.base import Base, TimestampMixin


class SysMenu(TimestampMixin, Base):
    __tablename__ = "sys_menu"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    parent_id: Mapped[int] = mapped_column(BigInteger)
    name: Mapped[str] = mapped_column(String(64))
    type: Mapped[str] = mapped_column(String(32))
    path: Mapped[str | None] = mapped_column(String(255), nullable=True)
    component: Mapped[str | None] = mapped_column(String(255), nullable=True)
    permission_code: Mapped[str | None] = mapped_column(String(128), nullable=True)
    icon: Mapped[str | None] = mapped_column(String(64), nullable=True)
    sort: Mapped[int] = mapped_column(Integer)
    visible: Mapped[int] = mapped_column(Integer)
    status: Mapped[int] = mapped_column(Integer)
    deleted: Mapped[int] = mapped_column(Integer)

